<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EditEntityDialog</class>
 <widget class="QDialog" name="EditEntityDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>372</width>
    <height>440</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Edit an entity</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QFormLayout" name="formLayout">
     <property name="fieldGrowthPolicy">
      <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
     </property>
     <item row="0" column="0">
      <widget class="QLabel" name="type_label">
       <property name="text">
        <string>Type</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLabel" name="type_field">
       <property name="font">
        <font>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Enemy</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="name_label">
       <property name="text">
        <string>Name</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLineEdit" name="name_field"/>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="layer_label">
       <property name="text">
        <string>Layer</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="EnumSelector&lt;Layer&gt;" name="layer_field"/>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="position_label">
       <property name="text">
        <string>Position</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <layout class="QHBoxLayout" name="horizontal_layout">
       <item>
        <widget class="QSpinBox" name="x_field">
         <property name="maximum">
          <number>999999</number>
         </property>
         <property name="singleStep">
          <number>8</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>,</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="y_field">
         <property name="maximum">
          <number>999999</number>
         </property>
         <property name="singleStep">
          <number>8</number>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontal_spacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="size_label">
       <property name="text">
        <string>Size</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <layout class="QHBoxLayout" name="size_field">
       <item>
        <widget class="QSpinBox" name="width_field">
         <property name="maximum">
          <number>999999</number>
         </property>
         <property name="singleStep">
          <number>8</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>x</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="height_field">
         <property name="maximum">
          <number>999999</number>
         </property>
         <property name="singleStep">
          <number>8</number>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontal_spacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item row="5" column="0">
      <widget class="QLabel" name="direction_label">
       <property name="text">
        <string>Direction</string>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <layout class="QHBoxLayout" name="horizontal_layout_2">
       <item>
        <widget class="QSpinBox" name="direction_field"/>
       </item>
       <item>
        <spacer name="horizontal_spacer_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item row="6" column="0">
      <widget class="QLabel" name="sprite_label">
       <property name="text">
        <string>Sprite</string>
       </property>
      </widget>
     </item>
     <item row="6" column="1">
      <widget class="ResourceSelector" name="sprite_field"/>
     </item>
     <item row="7" column="0">
      <widget class="QLabel" name="subtype_label">
       <property name="text">
        <string>Subtype</string>
       </property>
      </widget>
     </item>
     <item row="7" column="1">
      <widget class="QComboBox" name="comboBox"/>
     </item>
     <item row="8" column="0">
      <widget class="QLabel" name="treasure_label">
       <property name="text">
        <string>Treasure</string>
       </property>
      </widget>
     </item>
     <item row="8" column="1">
      <layout class="QVBoxLayout" name="treasure_field">
       <item>
        <widget class="ResourceSelector" name="treasure_name_field"/>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontal_layout_3">
         <item>
          <widget class="QLabel" name="treasure_variant_label">
           <property name="text">
            <string>Variant of this item</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="treasure_variant_field"/>
         </item>
         <item>
          <spacer name="horizontal_spacer_4">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QCheckBox" name="save_treasure_checkbox">
         <property name="text">
          <string>Save the treasure state</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontal_layout_4">
         <item>
          <widget class="QLabel" name="treasure_savegame_variable_label">
           <property name="text">
            <string>Savegame variable</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="treasure_savegame_variable_field"/>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="button_box">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
     <property name="centerButtons">
      <bool>false</bool>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ResourceSelector</class>
   <extends>QComboBox</extends>
   <header>gui/resource_selector.h</header>
  </customwidget>
  <customwidget>
   <class>EnumSelector&lt;Layer&gt;</class>
   <extends>QComboBox</extends>
   <header>layer_traits.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>button_box</sender>
   <signal>accepted()</signal>
   <receiver>EditEntityDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_box</sender>
   <signal>rejected()</signal>
   <receiver>EditEntityDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
